src/Orc.FileAssociation/Services/FileAssociationService.cs

```diff
@@ -7,44 +7,56 @@
 {
+    using System;
     using System.Runtime.InteropServices;
     using Catel;
     using Catel.Logging;
+    using Microsoft.Win32;
 
     public class FileAssociationService : IFileAssociationService
     {
         private static readonly ILog Log = LogManager.GetCurrentClassLogger();
 
-        public void AssociateFilesWithApplication(string applicationName = null)
+        public void AssociateFilesWithApplication(ApplicationInfo applicationInfo)
         {
-            applicationName = applicationName.GetApplicationName();
+            string applicationName = null;
+            if (applicationInfo.Name.GetApplicationName() is not null)
+            {
+               applicationName = applicationInfo.Name.GetApplicationName();
+            }
 
             Log.Info("Associating files with '{0}'", applicationName);
 
-            var applicationAssociationRegistrationUi = (IApplicationAssociationRegistrationUI)new ApplicationAssociationRegistrationUI();
-            var hr = applicationAssociationRegistrationUi.LaunchAdvancedAssociationUI(applicationName);
-            var exception = Marshal.GetExceptionForHR(hr);
-            if (exception != null)
+            foreach (var extension in applicationInfo.SupportedExtensions)
             {
-                Log.Error(exception, "Failed to associate the files with application '{0}'", applicationName);
-                throw exception;
+                var finalExtension = extension;
+                if (!finalExtension.StartsWith("."))
+                {
+                    finalExtension = "." + finalExtension;
+                }
+                var appPath = System.Diagnostics.Process.GetCurrentProcess().MainModule.FileName;
+                var key = Registry.CurrentUser.CreateSubKey("Software\\Classes\\" + finalExtension);
+                var subKey = key.CreateSubKey("shell\\open\\command");
+                subKey.SetValue("", appPath + " \"%1\"");
             }
 
             Log.Info("Associated files with '{0}'", applicationName);
         }
 
-        [Guid("1f76a169-f994-40ac-8fc8-0959e8874710")]
-        [InterfaceType(ComInterfaceType.InterfaceIsIUnknown)]
-        public interface IApplicationAssociationRegistrationUI
+        public void UndoAssociationFilesWithApplication(ApplicationInfo applicationInfo)
         {
-            [PreserveSig]
-            int LaunchAdvancedAssociationUI([MarshalAs(UnmanagedType.LPWStr)] string pszAppRegName);
-        }
+            foreach (var extension in applicationInfo.SupportedExtensions)
+            {
+                var finalExtension = extension;
+                if (!finalExtension.StartsWith("."))
+                {
+                    finalExtension = "." + finalExtension;
+                }
+
+                Microsoft.Win32.Registry.CurrentUser.DeleteSubKeyTree($"SOFTWARE\\Classes\\{finalExtension}");
+                Log.Debug($"Removing extension association {finalExtension} capabilities from current user");
```

Comment:
Removing => log *before* doing the operation
Removed => log *after* doing the operation

So this requires either the line to be moved before the registry action or change the wording.